.github/actions/build-ab/index.js

```diff
@@ -9,84 +10,131 @@ import { fetchRetry } from '../shared-utils/fetch-retry.js'
+const releasedScript = (await releasedScriptRequest.text()).replace(/\/\/# sourceMappingURL=.*?\.map/, '')
+const releasedScriptOutput = path.join(outputDir, `${args.environment}-released.js`)
+const releasedScriptTemplate = Handlebars.compile(await fs.promises.readFile(path.resolve(__dirname, './templates/released.js'), 'utf-8'))
+await fs.promises.writeFile(
+  releasedScriptOutput,
+  releasedScriptTemplate({
+    args, releasedScript
+  }),
+  { encoding: 'utf-8' }
+)
+fileList.push(releasedScriptOutput)
 
 if (['dev', 'staging'].includes(args.environment)) {
-  if (!args.abAppId || !args.abLicenseKey) {
-    throw new Error('Cannot deploy current loader or experiments without A/B app id and license key.')
-  }
+  // 2. Create latest environment script
+  const latestScriptRequest = await fetchRetry(`${args.latest}?_nocache=${uuidv4()}`, { retry: 3 })
 
-  const currentScript = await fetchRetry(`${args.current}?_nocache=${uuidv4()}`, { retry: 3 })
-  abScripts.push({ name: 'current', contents: await currentScript.text() })
-
-  const stsClient = new STSClient({ region: args.region })
-  const s3Credentials = await stsClient.send(new AssumeRoleCommand({
-    RoleArn: args.role,
-    RoleSessionName: 'downloadFromS3Session',
-    DurationSeconds: 900
-  }))
-
-  const s3Client = new S3Client({
-    region: args.region,
-    credentials: {
-      accessKeyId: s3Credentials.Credentials.AccessKeyId,
-      secretAccessKey: s3Credentials.Credentials.SecretAccessKey,
-      sessionToken: s3Credentials.Credentials.SessionToken
-    }
-  })
-
-  let isTruncated = true
-  const listCommand = new ListObjectsCommand({
-    Bucket: args.bucket,
-    Prefix: 'experiments/',
-    Delimiter: '/'
-  });
-  let experimentsList = new Set()
-
-  while (isTruncated) {
-    const { IsTruncated, NextMarker, CommonPrefixes } = await s3Client.send(listCommand)
-    if (CommonPrefixes) CommonPrefixes.forEach(prefix => experimentsList.add(prefix.Prefix))
-
-    if (IsTruncated) {
-      listCommand.input.Marker = NextMarker
-    } else {
-      isTruncated = false
-    }
+  if (!latestScriptRequest.ok() || typeof latestScriptRequest.text !== 'function') {
+    throw new Error('Could not retrieve the latest unpublished loader script.')
   }
 
-  if (experimentsList.size === 0) {
-    console.log('No experiments to include in A/B script.')
+  const latestScript = (await latestScriptRequest.text()).replace(/\/\/# sourceMappingURL=.*?\.map/, '')
+  const latestScriptOutput = path.join(outputDir, `${args.environment}-latest.js`)
+  const releasedScriptTemplate = Handlebars.compile(await fs.promises.readFile(path.resolve(__dirname, './templates/latest.js'), 'utf-8'))
+  await fs.promises.writeFile(
+    latestScriptOutput,
+    releasedScriptTemplate({
+      args, latestScript
+    }),
+    { encoding: 'utf-8' }
+  )
+  fileList.push(latestScriptOutput)
+
+  // 3. Creating experiments script
+  if (!args.role || !args.bucket || !args.region) {
```

Comment:
Can you please add a check/message for the env vars that AWS uses as well?

Reply:
There is a check in the action script. If the env variables are not set, it just skips the experiments. If I add it here, it will throw an error. We only ever need to connect to S3 when building for dev and staging environments.